### DISCUSSION TITLE
Format Document command doesn't organize imports with Biome extension

### DISCUSSION THREAD

**Discussion by batuhanoztrk** [biomejs/biome-vscode#424] [Q&A]
When using the Biome pre-release extension in VSCode, the "Format Document" command (`Ctrl/Cmd + Shift + P` > "Format Document" or `Alt + Shift + F`) doesn't organize imports automatically, even though organize imports is enabled in the Biome configuration.

### Environment
- VSCode Version: 1.93.1
- Biome Extension Version: 2024.10.131712 (pre-release)
- OS: Mac OS 15.1 (24B83)

### Configuration
My VSCode settings (`settings.json`):
~~~json
{
  "editor.accessibilitySupport": "off",
  "editor.unicodeHighlight.nonBasicASCII": false,
  "terminal.integrated.env.osx": {
    "FIG_NEW_SESSION": "1",
    "Q_NEW_SESSION": "1"
  },
  "workbench.colorTheme": "One Dark Pro Darker",
  "workbench.iconTheme": "a-file-icon-vscode",
  "editor.fontFamily": "Fira Code",
  "editor.fontLigatures": true,
  "editor.wordWrap": "on",
  "editor.fontSize": 13,
  "[typescriptreact]": {
    "editor.defaultFormatter": "biomejs.biome"
  },
  "[typescript]": {
    "editor.defaultFormatter": "biomejs.biome"
  },
  "[jsonc]": {
    "editor.defaultFormatter": "biomejs.biome"
  },
  "editor.formatOnSave": true,
  "[javascript]": {
    "editor.defaultFormatter": "biomejs.biome"
  },
  "files.autoSave": "afterDelay",
  "editor.formatOnType": true,
  "editor.formatOnPaste": true,
  "[json]": {
    "editor.defaultFormatter": "biomejs.biome",
  },
  "kotlin.languageServer.enabled": false,
  "kotlin.debugAdapter.enabled": false,
  "javascript.preferences.jsxAttributeCompletionStyle": "braces",
  "typescript.preferences.jsxAttributeCompletionStyle": "braces",
  "redhat.telemetry.enabled": true,
  "[dart]": {
    "editor.formatOnSave": true,
    "editor.formatOnType": true,
    "editor.rulers": [80],
    "editor.selectionHighlight": false,
    "editor.tabCompletion": "onlySnippets",
    "editor.wordBasedSuggestions": "off"
  },
  "javascript.updateImportsOnFileMove.enabled": "always",
  "editor.codeActionsOnSave": {
    "quickfix.biome": "explicit",
    "source.organizeImports.biome": "explicit"
  },
}
~~~

### Expected Behavior
When using the "Format Document" command, it should both format the code and organize imports.

### Actual Behavior
The "Format Document" command only formats the code but doesn't organize imports. I need to run the "Organize Imports" command separately.

### Additional Information
- The formatting itself works correctly
- Organize Imports works when run as a separate command
- This issue occurs specifically with the pre-release version of the extension

### Steps to Reproduce
1. Install Biome pre-release extension
2. Configure VSCode and Biome as shown above
3. Open a JavaScript/TypeScript file with unorganized imports
4. Run the "Format Document" command
5. Observe that imports remain unorganized

**Comment by whoop-t** [comment]
Biome format does not organize imports, you have to run biome check

Im not sure how this is configured in VSCode as I use something else, but I had this issue and had to make sure biome check was running

**Comment by ematipico** [comment]
The import sorting isn't part of formatting, and this is part of the LSP.

As you might notice the `source.organizeImport.biome` option is under the option `codeActionsOnSave`, this means that you trigger the import sorting when you save the file.

If you come from a different experience (prettier + plugin or eslint + plugin), they don't follow LSP, so that's why you have different expectations.

Biome is closer to LSP than other tools.
